app/controllers/registrations_controller.rb

```diff
@@ -1,8 +1,15 @@
 class RegistrationsController < ApplicationController
+  def create
+    if current_user
```

Comment:
Do we want to auto-log in the new user?  If so, I would do the user processing first and log them in if they aren't already.  then build the registration

Reply:
Added auto log in